### PR TITLE
Fix MSTEST0036 is shown for cases where no shadowing happens

### PR DESCRIPTION
**Pull request by engyebrahim**
fix: https://github.com/microsoft/testfx/issues/3824

### CODE REVIEW

**Review by Evangelink — COMMENTED** [review]
We have another FP when the methods are private (add the same private method in current and base and you will see a false positive).

**Review by Evangelink — APPROVED** [review]

File: src/Analyzers/MSTest.Analyzers/DoNotUseShadowingAnalyzer.cs [hunk 1]
```diff
@@ -112,7 +112,8 @@ private static bool IsMemberShadowing(ISymbol member, ISymbol baseMember)
         }
 
         // Compare methods
-        if (member is IMethodSymbol methodSymbol && baseMember is IMethodSymbol baseMethodSymbol)
+        if (member is IMethodSymbol methodSymbol && baseMember is IMethodSymbol baseMethodSymbol && methodSymbol.IsGenericMethod == baseMethodSymbol.IsGenericMethod
+            && !(methodSymbol.DeclaredAccessibility == Accessibility.Private && baseMethodSymbol.DeclaredAccessibility == Accessibility.Private))
```
Comment (cremor) [inline src/Analyzers/MSTest.Analyzers/DoNotUseShadowingAnalyzer.cs:116]: It also can't be shadowing if only one of the methods is private, or can it?

Comment (Evangelink) [reply src/Analyzers/MSTest.Analyzers/DoNotUseShadowingAnalyzer.cs:116]: You are right, I find the syntax confusing but @engyebrahim added the not before the parenthesis so the and clause becomes a or :)

Comment (cremor) [reply src/Analyzers/MSTest.Analyzers/DoNotUseShadowingAnalyzer.cs:116]: - private and private: `!(true && true) = false` (ok)
- public and public: `!(false && false) = true` (ok)
- private and public: `!(true && false) = true` (not ok, is not shadowing)

And I think the visibility check also needs to happen in the "Compare properties" section below.

Comment (Evangelink) [reply src/Analyzers/MSTest.Analyzers/DoNotUseShadowingAnalyzer.cs:116]: Oh yeah good point, I overlooked last case. @engyebrahim can you please add more tests and probably fix the remaining broken part?

Comment (engyebrahim) [reply src/Analyzers/MSTest.Analyzers/DoNotUseShadowingAnalyzer.cs:116]: public&public raises
public&private raises
image

private&private doesn't raise 
private&public doesn't raise -> `fixing that case`
image

**Pull request merged**